### ISSUE TITLE
While importing new beatmaps, randomizer can fail to select next song

### ISSUE THREAD

**Issue by LittleEndu** [ppy/osu#1905]
I was importing beatmaps from osu-stable while randomizing through songs on selection screen and noticed it sometimes failed to select next song and selected the same set, same difficulty instead.

**Comment by LittleEndu** [comment]
This should probably be moved to low priority instead of a milestone.

**Comment by LittleEndu** [comment]
Just saying, #2054 hasn't fixed this.

**Comment by peppy** [comment]
It wasn't intended to, and does not deal with this code at all.

**Comment by LittleEndu** [comment]
I was hoping the new edits have made importing beatmaps more stable but I got it to crash instead while the same issue can now occur multiple times in the row while it used to happened only once. 

[runtime.log](https://github.com/ppy/osu/files/2080650/runtime.log)

**Comment by peppy** [comment]
Looks like a cross-thread issue. Bumping to fix for this month.

**Comment by Joehuu** [comment]
Related to #4818.

**Comment by Morilli** [comment]
Fixed by #4872 I'm assuming, because I can no longer replicate this and it makes absolute sense for the referencing PR to also have fixed this.
